### PR TITLE
New rule: Unused kotlin extensions

### PR DESCRIPTION
**Pull request by tasomaniac**
Kotlin Android Extensions are officially deprecated and will be completely removed as part of Kotlin 1.8

This new rule checks if it is unused and can be removed from a module

### CODE REVIEW

File: modulecheck-core/src/main/kotlin/modulecheck/core/UnusedPluginFinding.kt [hunk 1]
```diff
@@ -22,27 +22,26 @@ import modulecheck.api.finding.Fixable
...
-import modulecheck.core.rule.KAPT_PLUGIN_FUN
-import modulecheck.core.rule.KAPT_PLUGIN_ID
 import modulecheck.parsing.gradle.Declaration
 import modulecheck.project.McProject
 import modulecheck.utils.LazyDeferred
 import modulecheck.utils.lazyDeferred
 import java.io.File
 
-data class UnusedKaptPluginFinding(
+data class UnusedPluginFinding(
```
Comment (tasomaniac) [inline modulecheck-core/src/main/kotlin/modulecheck/core/UnusedPluginFinding.kt:31]: Made this generic. Can be used with any unused Gradle plugin

**Comment by tasomaniac** [comment]
The `versioning` CI workflow failed but I'm not sure if that is related to the PR.

**Review by RBusarow — APPROVED** [review]
Thanks for this!

Don't forget that the legacy plugin also handles `@Parcelize`, which fortunately has a different namespace than the current version.

I added a check for that reference to the rule, and added a test to ensure that rule doesn't give a false positive for "unused" when that older annotation is referenced.

File: modulecheck-core/src/main/kotlin/modulecheck/core/UnusedPluginFinding.kt (hunk 1 above)
Comment (RBusarow) [reply modulecheck-core/src/main/kotlin/modulecheck/core/UnusedPluginFinding.kt:31]: This is good!

The kapt-oriented stuff is just about the oldest code in the project, and largely unchanged.  It doesn't quite fit the patterns of everything else, but it hasn't been bad enough for me to revisit it just yet.

At some point, I'm going to want to broaden the somewhat-dated KAPT checks to a more generic "code generation" scope which includes Anvil and KSP extensions.

So, this is a step in that direction. :)

**Comment by tasomaniac** [comment]
Thanks. It is great that GitHub now allows you to push into forked repos when PR is open. This was a challenge in the past.

The Parcelize is there but isn't that now a separate plugin? If synthetic imports are not used, this plugin should be removed. It should be replaced by Parcelize specific one

**Comment by RBusarow** [comment]
The two plugins use different annotation classes for `@Parcelize`.

old: `kotlinx.android.parcel.Parcelize`
new: `kotlinx.parcelize.Parcelize`

So it could technically still be auto-fixed by replacing the plugin, but it would also require updating the Kotlin source.

**Pull request merged**